components/Typography/tests/unit/MaterialScalableFontTests.m

```diff
@@ -0,0 +1,266 @@
+  UIFont *scalableFont2 = [scaler2 scalableFontWithFont:font];
+
+  UIFont *scaledFont1 = [scalableFont1 mdc_scaledFontAtDefaultSize];
+  UIFont *scaledFont2 = [scalableFont2 mdc_scaledFontAtDefaultSize];
+
+  // Then
+  XCTAssert([scaledFont1 mdc_isSimplyEqual:scaledFont2]);
+}
+
+- (void)testNonScaledFontReturnsSelf {
+  // Given
+  UIFont *font = [UIFont systemFontOfSize:18.0];
+
+  // Note that no scaling curve has been attached to the font, so it will NOT scale
+  UIFont *nonScaledFont1 = [font mdc_scaledFontForSizeCategory:UIContentSizeCategoryExtraLarge];
+
+  // Then
+  XCTAssert([font mdc_isSimplyEqual:nonScaledFont1]);
+}
+
+- (void)testNegativeAndZeroScalingCurve {
+  // Given
+  UIFont *font = [UIFont systemFontOfSize:18.0];
+
+  NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+    UIContentSizeCategoryExtraSmall : @0,
+    UIContentSizeCategorySmall : @0,
+    UIContentSizeCategoryMedium : @0,
+    UIContentSizeCategoryLarge : @0,
+    UIContentSizeCategoryExtraLarge : @0,
+    UIContentSizeCategoryExtraExtraLarge : @0,
+    UIContentSizeCategoryExtraExtraExtraLarge : @0,
+    UIContentSizeCategoryAccessibilityMedium : @-1,
+    UIContentSizeCategoryAccessibilityLarge : @-1,
+    UIContentSizeCategoryAccessibilityExtraLarge : @-1,
+    UIContentSizeCategoryAccessibilityExtraExtraLarge : @-1,
+    UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @-1,
+  };
+
+  font.mdc_scalingCurve = scalingCurve;
+
+  // Note that scaling curve is 0 @ UIContentSizeCategoryExtraExtraExtraLarge so this font should be
+  // the same size.
+  UIFont *zeroScaledFont =
+      [font mdc_scaledFontForSizeCategory:UIContentSizeCategoryExtraExtraExtraLarge];
+
+  // Note that scaling curve is -1 @ UIContentSizeCategoryAccessibilityExtraExtraExtraLarge so this
+  // font should be the same size.
+  UIFont *negativeScaledFont =
+      [font mdc_scaledFontForSizeCategory:UIContentSizeCategoryAccessibilityExtraExtraExtraLarge];
+
+  // Then
+  XCTAssert([font mdc_isSimplyEqual:zeroScaledFont]);
+  XCTAssert([font mdc_isSimplyEqual:negativeScaledFont]);
+}
+
+- (void)testIncompleteScalingCurve {
+  // Given
+  UIFont *font = [UIFont systemFontOfSize:20.0];
+
+  const CGFloat curvePointSize = 12.0;
+
+  // This curve is missing all values over Medius
+  NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+    UIContentSizeCategoryExtraSmall : @(curvePointSize),
+    UIContentSizeCategorySmall : @(curvePointSize),
+    UIContentSizeCategoryMedium : @(curvePointSize),
+  };
+
+  font.mdc_scalingCurve = scalingCurve;
+
+  UIFont *mediumScaledFont = [font mdc_scaledFontForSizeCategory:UIContentSizeCategoryMedium];
+
+  UIFont *missingCurveScaledFont =
+      [font mdc_scaledFontForSizeCategory:UIContentSizeCategoryAccessibilityExtraExtraExtraLarge];
+
+  // Then
+  XCTAssertTrue(MDCCGFloatEqual(mediumScaledFont.pointSize, curvePointSize));
+  XCTAssert([font mdc_isSimplyEqual:missingCurveScaledFont]);
```

Comment:
Shouldn't this be `XCTAssertEquals`? I would expect that `self` is returned for a missing curve.  Ideally, an NSException is raised, but returning self seems expected given  the current behavior.

Reply:
Spoke offline.